File: LeetCode/RemoveDuplicatesFromSortedArray.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        last = "NaN"
        uniques = 0
        i = 0
        while i < len(nums): # len must be evaluated at every step for this to work
            if nums[i] == last:
                del nums[i]
            else:
                uniques += 1
                last = nums[i]
                i += 1
        return uniques

File: LeetCode/RemoveDuplicatesFromSortedArray.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() == 0) {
            return 0;
        }
        int last_val = nums[0];
        int last_val_i = 0;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] != last_val) {
                nums[last_val_i + 1] = nums[i];
                last_val = nums[i];
                last_val_i++;
            }
        }
        return last_val_i + 1;
    }
};
